### PR TITLE
Infrastructure: Upgrade to ESLint 9

### PR DESCRIPTION
**Pull request by nschonni**
Took a pass at converting to the new flat config and ESLint 9.
- Dropped Prettier plugins and replaced the with updated globs
- Hacked around Ava's plugin not officially supporting v9 yet
- Prettier ignore for a conflicting formatting
- Bump ES6 to ES2021 since it was already used by app.js
___
[WAI Preview Link](https://deploy-preview-320--aria-practices.netlify.app/ARIA/apg) _(Last built on Tue, 04 Jun 2024 18:52:32 GMT)._

### CODE REVIEW

**Comment by nschonni** [comment]
@howard-e I gave this a rebase for the conflicts

**Comment by howard-e** [comment]
Okay cool! Will review

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        },
+      },
+      globals: {
+        ...globals.browser,
+      },
+      sourceType: 'script',
+    },
+
+    linterOptions: {
+      reportUnusedDisableDirectives: true,
```
Comment (howard-e) [inline eslint.config.mjs:39]: Or ['warn'](https://eslint.org/docs/latest/use/configure/migration-guide#linter-options) if wanting to be explicit. According to [Reporting Unused Disable Directives](https://eslint.org/docs/latest/use/configure/configuration-files#reporting-unused-disable-directives):

> For legacy compatibility, true is equivalent to "warn" ...

File: eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import html from 'eslint-plugin-html';
+import ava from 'eslint-plugin-ava';
+
+export default [
+  // Replaces .eslintignore
+  {
+    ignores: [
+      '**/*.min.js',
+      'common/',
+      'content/patterns/landmarks/examples/js/visua11y.js', // cspell:ignore visua
```
[the remark this answers is not in the document]
Comment (nschonni) [reply eslint.config.mjs:16]: words "break" on the number, so cSpell is seeing "visua" + "11y". The comment is to ignore that non-typo

File: eslint.config.mjs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      'content/shared/js/highlight.pack.js',
+      'content/shared/js/skipto.js',
+    ],
+  },
+  js.configs.recommended,
+  {
+    files: ['**/*.js'],
+
+    languageOptions: {
+      ecmaVersion: 2021,
```
[the remark this answers is not in the document]
Comment (nschonni) [reply eslint.config.mjs:26]: I don't think so, I think I was probably copying to the old `11` value, but I guess that would actually be `2020` https://archive.eslint.org/docs/user-guide/configuring/language-options

File: eslint.config.mjs (hunk 1 above)
Comment (nschonni) [reply eslint.config.mjs:39]:
```suggestion
      reportUnusedDisableDirectives: 'error',
```

File: eslint.config.mjs (hunk 2 above)
[the remark this answers is not in the document]
Comment (howard-e) [reply eslint.config.mjs:16]: Oh I see!

File: eslint.config.mjs (hunk 3 above)
[the remark this answers is not in the document]
Comment (howard-e) [reply eslint.config.mjs:26]: Okay got it, think latest would still be fine here. But it wouldn't make a change to anything right now. This ok to me

**Review by howard-e — APPROVED** [review]
> Took a pass at converting to the new flat config and ESLint 9.

@nschonni this is great, so much more than just a "pass" -- thanks!

**Pull request merged**